### DISCUSSION TITLE
Can you optimize the style of the interface?

### DISCUSSION THREAD

**Discussion by 1203086522** [jsme-editor/jsme-editor.github.io#16] [Ideas]
I really hope you can optimize the icons and interface, because it looks very old.  Or let the user customize the appearance

**Comment by peter-ertl** [comment]
Hello,
could you provide some suggestions how the better and more modern interface
should look like?

**Reply by 1203086522** [reply]
Thank you for your reply. My thought is whether the icons of these buttons can be customized by users (such as iconfont icons), and whether the activation style of the buttons and the lines of the borders can be adjusted by users (now you seem to be using sprites)
![image](https://user-images.githubusercontent.com/51741106/131597735-da44320a-77bd-4ead-8081-7683fde3118b.png)

**Comment by 1203086522** [comment]
I found some editors that look good on the Internet, hoping to provide you with some optimization inspiration:

![image](https://user-images.githubusercontent.com/51741106/131603275-1fc7fc23-4aa3-4bca-9fc5-bd4294f06fe7.png)

![image](https://user-images.githubusercontent.com/51741106/131602841-36d37c93-f4f1-469a-bbc4-22c6276affc6.png)

![image](https://user-images.githubusercontent.com/51741106/131603182-1dc5fbef-ffed-4b69-ac0f-3eba32863617.png)

![image](https://user-images.githubusercontent.com/51741106/131603199-99f7316d-1864-4b78-9bcb-c2fc523e68d4.png)

![image](https://user-images.githubusercontent.com/51741106/131603220-5246de4c-e34b-4e87-9d67-2c59e1c58b28.png)

![image](https://user-images.githubusercontent.com/51741106/131603258-75dc666e-95b3-46fd-9f62-23a68ce636e7.png)

The distance between their icon and the sideline seems to be larger

**Reply by 1203086522** [reply]
In terms of layout: (I am not sure whether the following method will be more beautiful, just for reference)

![image](https://user-images.githubusercontent.com/51741106/131603725-26d63c0a-2219-446a-9447-d27cd118cdd7.png)

![image](https://user-images.githubusercontent.com/51741106/131603747-4b9d4fa4-bb0c-41bc-a6c0-6e0a3b5d1fc3.png)

**Comment by jsme-editor** [comment]
> the icons of these buttons can be customized by users (such as iconfont icons), and whether the activation style of the buttons and the lines of the borders can be adjusted by users (now you seem to be using sprites)

Yes JSME, looks old. JSME is based on JME, the more than 20 years Java editor. The code of the user interface is still the same. It is implement using the Java AWT GUI toolkit. There are no icons. All buttons are drawn using the java.awt.Graphics2D class. In JSME, the graphics 2D are converted to SVG (most recent browser) or  VML (old Internet Explorer browsers). Thus, reimplementing the user interface to support button customization would require some work. There are no sprites in JSME.

Customization of the line and borders might be easier to implement.

> In terms of layout: (I am not sure whether the following method will be more beautiful, just for reference)

Options to reorganize the layout  might be easier to implement.

> The distance between their icon and the sideline seems to be large

Yes that looks nicer, but uses also more space.


Thank you for your suggestions.

Bruno Bienfait
